config/config.go

```diff
@@ -573,9 +575,31 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		tiNames[mt.Name] = struct{}{}
 	}
 
+	if c.GoogleCloud.ExternalURL != nil {
+		if err := ExternalURLNormalize(c.GoogleCloud.ExternalURL.URL); err != nil {
+			return err
+		}
+	}
+
 	return checkTimeInterval(c.Route, tiNames)
 }
 
+// ExternalURLNormalize normalizes the external URL, validating the scheme
+// and ensuring there's no trailing slash.
+func ExternalURLNormalize(u *url.URL) error {
```

Comment:
suggestion: add a corresponding unit test to config/config_test.go

To ensure `ExternalURLNormalize` does what you expect.

suggestion: can be a private function `externalURLNormalize`, no?

Reply:
Created a unit test. `ExternalURLNormalize` cannot be a private method because we use it in the `config` package.